### DISCUSSION TITLE
build with program with vcpkg

### DISCUSSION THREAD

**Discussion by warsark** [microsoft/vcpkg#33791] [General]
vcpkg install llvm[clang]:x86-windows-static
vcpkg install llvm[clang]:x64-windows-static
VS2022 Create a console program(x86 console build) with integrate vcpkg
build the console,the output:
1>LINK : the 32-bit linker (C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX86\x86\link.exe) failed to do memory mapped file I/O on `F:\vcpkg\installed\x86-windows-static\debug\lib\clangStaticAnalyzerCheckers.lib' and is going to restart linking with a 64-bit linker for better throughput
1>LINK : restarting link with 64-bit linker `C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x86\link.exe'

why?

**Comment by Neumann-A** [comment]
Building LLVM requires a lot of RAM so the 32 bit linker is probably reaching the 2,5 GB memory limit x86 has. Don't know why VS/vcpkg is not using the `HostX64` tools by default

**Comment by warsark** [comment]
but my program is a console template create with vs2022,it not comtains any LLVM reference

**Reply by autoantwort** [reply]
But you have installed `llvm[clang]` and you use the vcpkg integration so your program gets automatically linked against llvm.

**Reply by Neumann-A** [reply]
another reason to avoid the msbuild integration ;)
